src/main/kotlin/com/dansplugins/factionsystem/command/faction/addmember/MfFactionAddMemberCommand.kt

```diff
@@ -0,0 +1,110 @@
+package com.dansplugins.factionsystem.command.faction.addmember
+
+import com.dansplugins.factionsystem.MedievalFactions
+import com.dansplugins.factionsystem.faction.MfFaction
+import com.dansplugins.factionsystem.faction.MfFactionMember
+import com.dansplugins.factionsystem.player.MfPlayer
+import dev.forkhandles.result4k.onFailure
+import org.bukkit.ChatColor
+import org.bukkit.command.Command
+import org.bukkit.command.CommandExecutor
+import org.bukkit.command.CommandSender
+import org.bukkit.command.TabCompleter
+import org.bukkit.entity.Player
+import preponderous.ponder.command.dropFirst
+import java.util.logging.Level
+
+class MfFactionAddMemberCommand(private val plugin: MedievalFactions) : CommandExecutor, TabCompleter {
+
+    override fun onCommand(sender: CommandSender, command: Command, label: String, args: Array<out String>): Boolean {
+        if (!sender.hasPermission("mf.force.addmember")) {
+            sender.sendMessage("${ChatColor.RED}${plugin.language["CommandFactionAddMemberNoPermission"]}")
+            return true
+        }
+        if (sender !is Player) {
+            sender.sendMessage("${ChatColor.RED}${plugin.language["CommandFactionAddMemberNotAPlayer"]}")
+            return true
+        }
+        if (args.isEmpty()) {
+            sender.sendMessage("${ChatColor.RED}${plugin.language["CommandFactionAddMemberUsage"]}")
+            return true
+        }
+
+        // get target player
+        val targetPlayer = plugin.server.getOfflinePlayer(args[0])
+        if (!targetPlayer.hasPlayedBefore()) {
```

Comment:
You may also wish to check if the target is online here - hasPlayedBefore returns false in their first play session

Reply:
I've added that check to cover the case where players are on their first play session.